File: AN_2/SEM_1/MAP/lab2/src/decorator/DelayTaskRunner.java
package decorator;

public class DelayTaskRunner extends AbstractTaskRunner
{
    public DelayTaskRunner(TaskRunner taskRunner)
    {
        super(taskRunner);
    }

    @Override
    public void executeOneTask()
    {
        super.executeOneTask();
        try
        {
            Thread.sleep(1500);
        }
        catch (InterruptedException e)
        {
            e.printStackTrace();
        }
    }
}

File: AN_2/SEM_1/MAP/lab9/lab9/Decorator/DelayTaskRunner.cs
using System;
using System.Threading;

namespace Decorator
{
    public class DelayTaskRunner : AbstractTaskRunner
    {
        public DelayTaskRunner(ITaskRunner taskRunner)
            : base(taskRunner)
        {
        }

        public override void ExecuteOneTask()
        {
            base.ExecuteOneTask();
            try
            {
                Thread.Sleep(1500);
            }
            catch (ThreadInterruptedException e)
            {
                Console.WriteLine(e.Message);
            }
        }
    }
}
